library/core/src/slice/iter.rs

```diff
@@ -16,6 +16,9 @@ use crate::ptr::{self, without_provenance, without_provenance_mut, NonNull};
 
 use super::{from_raw_parts, from_raw_parts_mut};
 
+#[stable(feature = "boxed_slice_into_iter", since = "CURRENT_RUSTC_VERSION")]
+impl<T> !Iterator for [T] {}
```

Comment:
I don't think this is controversial, but why is this needed?

Reply:
Because otherwise we have an overlap with the `I: IntoIterator where I: Iterator` impl, via the `Box<I>: Iterator where I: Iterator` impl. I believe it has to do with the fact that `Box<I, A>` is fundamental, and it can't figure out that `Box<[I], A>` cannot be impl'd by a downstream crate. (I think it's the `A` parameter that it thinks can be substituted further with a downstream type...)

Reply:
Yep, it's definitely the allocator param.

Reply:
You're correct in the analysis here; this is what I had to add to make the library impl work